SIPS/sip-32.md

```diff
@@ -0,0 +1,89 @@
+---
+sip: 32
+title: Add `snap_trackEvent` method for pre-installed Snaps
+status: Draft
+author: Daniel Rocha (@danroc)
+created: 2025-03-20
+---
+
+## Abstract
+
+This Snap Improvement Proposal (SIP) introduces a new method,
+`snap_trackEvent`, allowing pre-installed Snaps to submit tracking events
+through the client. The client will determine how to handle these events.
+
+This feature enables Snaps to utilize the existing MetaMask client
+infrastructure for event tracking while maintaining user privacy.
+
+## Motivation
+
+Currently, there is no standardized way for pre-installed Snaps to submit
+analytics or tracking events. This proposal aims to:
+
+- Enable pre-installed Snaps to reuse MetaMask’s event tracking infrastructure.
+
+- Allow the MetaMask client to control event handling, ensuring security and
+  compliance with privacy policies.
+
+## Specification
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written
+in uppercase in this document are to be interpreted as described in [RFC
+2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### `snap_trackEvent`
+
+The `snap_trackEvent` method allows pre-installed Snaps to submit structured
+tracking events to the client. It is the client’s responsibility to process
+these events accordingly.
+
+The client MUST ensure that only pre-installed Snaps can call this method.
```

Comment:
Is there some endowment that guarantees this? Is the endowment only available to pre-installed Snaps? I think we need to document how this is enforced.

Reply:
I think it's an implementation detail _how_ the client enforces it.

Reply:
Not necessarily, it is important to include in the spec if a new permission is required.

That being said, I think we could make this available to all preinstalled Snaps without a permission/endowment.

WDYT @Mrtenz?

Reply:
> That being said, I think we could make this available to all preinstalled Snaps without a permission/endowment.

I agree, I don't see a reason to make this permissioned.

Reply:
For the specs sake you can just add something like
```
`snap_trackEvent` is endowed on any Snap that the client considers preinstalled.
```

Similar to https://metamask.github.io/SIPs/SIPS/sip-30